ion-schema/src/ion_path.rs

```diff
@@ -0,0 +1,90 @@
+use ion_rs::value::owned::Element;
+use ion_rs::value::Builder;
+use ion_rs::Symbol;
+use std::fmt;
+use std::fmt::Formatter;
+
+#[derive(Clone, PartialEq)]
+pub enum IonPathElement {
+    IndexedElement { index: usize },
+    Field { name: String },
+}
+
+// TODO: This can be removed once we have a complete Display for violation
+// This is currently used by schema sandbox to print nested violations
+impl fmt::Debug for IonPathElement {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self {
+            IonPathElement::IndexedElement { index } => {
+                write!(f, "{index}")
+            }
+            IonPathElement::Field { name } => {
+                write!(f, ".{name}")
+            }
+        }
+    }
+}
+
+impl fmt::Display for IonPathElement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match &self {
+            IonPathElement::IndexedElement { index } => {
+                write!(f, "[{index}]")
+            }
+            IonPathElement::Field { name } => {
+                write!(f, ".{name}")
+            }
+        }
+    }
+}
+
+#[derive(Default, Clone, PartialEq)]
```

Comment:
I don't know if we can derive it, but I think implementing `PartialOrd` might be useful for customers who want to extract all of the violation paths to create a flattened list of violations.
* if you want to confirm that a particular piece of data has exactly violations X, Y, and Z;  instead of having to move the violations into a set, you could just compare the sorted violations
* If you want to display a list of violations to a user, it will be easier for a human to read them if they have some logical ordering